src/shared/edit-util.c

```diff
@@ -104,37 +105,33 @@ int edit_files_add(
         return 1;
 }
 
-static int create_edit_temp_file(
-                const char *target_path,
-                const char *original_path,
-                char * const *comment_paths,
-                const char *marker_start,
-                const char *marker_end,
-                char **ret_temp_filename,
-                unsigned *ret_edit_line) {
-
+static int create_edit_temp_file(EditFile *e) {
         _cleanup_free_ char *temp = NULL;
         unsigned line = 1;
         int r;
 
-        assert(target_path);
-        assert(!comment_paths || (marker_start && marker_end));
-        assert(ret_temp_filename);
+        assert(e);
+        assert(e->context);
+        assert(e->path);
+        assert(!e->comment_paths || (e->context->marker_start && e->context->marker_end));
+
+        if (isempty(e->temp))
+                return 0;
```

Comment:
Hmm, I think this is wrong. At this point `e->temp` is usually empty, and we should return early if it's not empty, right?

Reply:
Ouch. The condition was inverted...